docs/guides/go-built-in.md

```diff
@@ -311,112 +310,96 @@ information on why the transaction was rejected.
+    for i, tx := range req.Txs {
+        if code := app.isValid(tx); code != 0 {
+            log.Printf("Error: invalid transaction index %v", i)
+            txs[i] = &abcitypes.ExecTxResult{Code: code}
+        } else {
+            parts := bytes.SplitN(tx, []byte("="), 2)
+            key, value := parts[0], parts[1]
+            log.Printf("Adding key %s with value %s", key, value)
+
+            if err := app.onGoingBlock.Set(key, value); err != nil {
+                log.Panicf("Error writing to database, unable to execute tx: %v", err)
+            }
+
+            log.Printf("Successfully added key %s with value %s", key, value)
+
+            // Marshal the ExecTxResult to the txs slice element
+            txs[i] = &abcitypes.ExecTxResult{}
+            if _, err := txs[i].MarshalToSizedBuffer(nil); err != nil {
+                log.Panicf("Error marshalling tx result: %v", err)
+            }
+        }
+    }
+
+    return &abcitypes.ResponseFinalizeBlock{
+    TxResults:        txs,
+    AppHash:          []byte{},
+    Events:           []abcitypes.Event{},
+    ValidatorUpdates: abcitypes.ValidatorUpdates{},
+    }, nil
 }
 ```
 
-Note that we check the validity of the transaction _again_ during `DeliverTx`.
-Transactions are not guaranteed to be valid when they are delivered to an
-application, even if they were valid when they were proposed.
-This can happen if the application state is used to determine transaction
-validity. Application state may have changed between the initial execution of `CheckTx`
-and the transaction delivery in `DeliverTx` in a way that rendered the transaction
-no longer valid.
-
-`EndBlock` is called to inform the application that the full block has been delivered
-and give the application a chance to perform any other computation needed, before the
-effects of the transactions become permanent.
-
-Note that `EndBlock` **cannot** yet commit the Badger transaction we were building
-in during `DeliverTx`.
-Since other methods, such as `Query`, rely on a consistent view of the application's
-state, the application should only update its state by committing the Badger transactions
-when the full block has been delivered and the `Commit` method is invoked.
-
 The `Commit` method tells the application to make permanent the effects of
 the application transactions.
 Let's update the method to terminate the pending Badger transaction and
 persist the resulting state:
 
 ```go
-func (app *KVStoreApplication) Commit() abcitypes.ResponseCommit {
-	if err := app.onGoingBlock.Commit(); err != nil {
-		log.Panicf("Error writing to database, unable to commit block: %v", err)
-	}
-	return abcitypes.ResponseCommit{Data: []byte{}}
+func (app KVStoreApplication) Commit(_ context.Context, commit *abcitypes.RequestCommit) (*abcitypes.ResponseCommit, error) {
+    if err := app.onGoingBlock.Commit(); err != nil {
+        return &abcitypes.ResponseCommit{}, err
+    }
+    return &abcitypes.ResponseCommit{}, nil
```

Comment:
```suggestion
    return &abcitypes.ResponseCommit{}, app.onGoingBlock.Commit()
```

Even if both are equivalent (CometBFT will panic anyway if `err` is not `nil`), the old version is easier to follow for a beginner.